proxy/api/ts/ts.h

```diff
@@ -1481,6 +1481,9 @@ tsapi TSReturnCode TSHttpTxnParentProxyGet(TSHttpTxn txnp, const char **hostname
  */
 tsapi void TSHttpTxnParentProxySet(TSHttpTxn txnp, const char *hostname, int port);
 
+tsapi TSReturnCode TSHttpTxnParentSelectionUrlGet(TSHttpTxn txnp, TSMBuffer bufp, TSMLoc obj);
+tsapi TSReturnCode TSHttpTxnParentSelectionUrlSet(TSHttpTxn txnp, TSMBuffer bufp, TSMLoc obj);
```

Comment:
There's definitely a real use case where you want to do parent selection on a different (normalized) URL that e.g. the pristine URL or cache key URL. Doing parent selection on the cache key URL would be a benefit (think of e.g. URLs with signatures, where you don't want the parent selection to include that signature).